roadmap/implementors-guide/src/node/utility/provisioner.md

```diff
@@ -38,6 +38,12 @@ At initialization, this subsystem has no outputs. Block authors can send a `Prov
 
 Note that block authors must re-send a `ProvisionerMessage::RequestBlockAuthorshipData` for each relay parent they are interested in receiving provisionable data for.
 
+## Block Production
+
+When a validator is selected by BABE to author a block, it becomes a block producer. The provisioner is the subsystem best suited to choosing which specific backed candidates and availability bitfields should be assembled into the block. To engage this functionality, a `ProvisionerMessage::RequestInherentData` is sent; the response is a set of non-conflicting candidates and the appropriate bitfields. Non-conflicting generally means that there are never two distinct parachain candidates included for the same parachain.
+
+One might ask: given `ProvisionerMessage::RequestInherentData`, what's the point of `ProvisionerMessage::RequestBlockAuthorshipData`? The answer is that the block authorship data includes more information than is present in the inherent data; disputes, for example.
```

Comment:
Wouldn't any consumer of `RequestInherentData` be interested in disputes as well?

Reply:
In practice our main consumer is the block authorship logic, which is clearly interested in disputes so it can report them to the chain. I am not aware of any other component which would be interested in disputes or dispute-related reports.

Reply:
The proposer consumes `RequestInherentData`, and if it cares about disputes, I haven't yet seen why it does.

Reply:
Yeah, on further inspection we probably don't need the proposer to be aware of disputes. I was thinking about a more antiquated approach where the `Proposer` constructs the dispute transactions and submits them to the block directly.

It seems that the "right way" to submit reports (based on @andresilva's work in GRANDPA/BABE) is to call a runtime API that submits a transaction to the queue. I guess that will end up being the main role of the MA subsystem - acting as an intermediary between POD reports and the Runtime API subsystem. Maybe we should rename it to Misbehavior Reporting, as most reports are "complete", and the actual dispute resolution / arbitration process will happen on-chain in the `Validity` module.

Reply:
Then we shouldn't need the dispute records to be present in the `Provisioner` then either, because the assumption is that anything that the Provisioner is responsible for everything that the Proposer will need. If the Proposer doesn't need them, then the Provisioner shouldn't either. And then I suggest removing `RequestBlockAuthorshipData`, as I don't see an immediate need for it if there are no disputes and no other intended users of the `Provisioner`.

Reply:
I think I agree with those proposals, but I also think they're out of scope for this PR.

Reply:
Yup, that's fair. We have a discussion on-record to reference now at least

Reply:
Created an issue from this so we don't lose it: #1362.